ql/time/calendar.hpp

```diff
@@ -240,6 +246,14 @@ namespace QuantLib {
         return impl_->isBusinessDay(_d);
     }
 
+    inline bool Calendar::isStartOfMonth(const Date& d) const {
+        return (d.month() != adjust(d-1, Preceding).month());
```

Comment:
I wonder if
```
        return d == startOfMonth(d);
```
would be more or less efficient?  It would be more clear for sure, and we wouldn't have two different implementations for two very related methods.  The same goes for `isEndOfMonth`.

Reply:
We could do that, but would the test at https://github.com/lballabio/QuantLib/blob/6331e50779f2c50d0294492e084cedae47e5c986/test-suite/calendars.cpp#L3383 then really test anything?

Reply:
It would still test consistency of the two methods, as if we were unaware of the implementation (which may change).